src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/action/AllocationActionConfig.kt

```diff
@@ -61,12 +63,36 @@ data class AllocationActionConfig(
             .endObject()
     }
 
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
+        require = suppressWarning(sin.readMap()),
+        include = suppressWarning(sin.readMap()),
+        exclude = suppressWarning(sin.readMap()),
+        waitFor = sin.readBoolean(),
+        index = sin.readInt()
+    )
+
+    @Throws(IOException::class)
+    override fun writeTo(out: StreamOutput) {
+        super.writeTo(out)
+        out.writeMap(require)
+        out.writeMap(include)
+        out.writeMap(exclude)
+        out.writeBoolean(waitFor)
+        out.writeInt(index)
```

Comment:
I believe the action index is already being written in the ActionConfig class which you're calling with super, so shouldn't need to write and read on all subclassed action configs.

Reply:
yes, it's duplicate, but it's needed in ActionConfig fromStreamInput